mongodbatlas/resource_mongodbatlas_cluster.go

```diff
@@ -1433,6 +1433,10 @@ func expandProcessArgs(d *schema.ResourceData, p map[string]interface{}) *matlas
 		}
 	}
 
+	if _, ok := d.GetOkExists("advanced_configuration.0.oplog_min_retention_hours"); ok {
+		res.OplogMinRetentionHours = pointy.Float64(cast.ToFloat64(p["oplog_min_retention_hours"]))
```

Comment:
If value is optional then it can be invalid. For that ToFloat returns error and zero as value. 
cast.ToFloat64 would not work properly with structure fields having omitempty as we want to set api object field as nil? Does that make sense?

```suggestion
                 oplog, err := cast.ToFloat64(p["oplog_min_retention_hours"])
                 // Optional: return error to end users that value is wrong
                 if err!=nil 
		    return diag.FromErr(errors.New("oplog_min_retention_hours"))
		}else{
		    res.OplogMinRetentionHours = pointy.Float64(oplog)
		}
```